### PR TITLE
change logrus to zerolog

### LINKED ISSUE #292: Replace logrus with zero log
### Issue To Be Solved

logrus is in maintenance mode, they even suggest zerolog, maybe we can migrate to zerolog too (fvm uses that already)

### PR DESCRIPTION
**Pull request by bluesign**
Closes #292

## Description

if we decide to remove logrus and replace with zerolog. 

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by bluesign** [comment]
@sideninja what do you think about this?

**Comment by sideninja** [comment]
Yes pls

**Comment by bluesign** [comment]
super this will help with realtime showing logs also, especially needed for debugging. rebasing to master.

**Comment by sideninja** [comment]
>  rebasing to master.

Why rebasing? let's be careful to not mess the master by rebasing.

**Comment by sideninja** [comment]
Oh you mean the other branch rebased?

File: server/storage.go [hunk 1]
```diff
@@ -122,12 +122,6 @@ func (s *BadgerStorage) Start() error {
 				return errors.Wrap(err, "failed to perform garbage collection on Badger DB")
 			}
 
-			s.logger.
-				WithFields(logrus.Fields{
-					"interval":     s.gcInterval,
-					"discardRatio": s.gcDiscardRatio,
-				}).
-				Debug("Performed garbage collection on Badger value log")
```
Comment (janezpodhostnik) [inline server/storage.go:130]: this one disappeared altogether, is that ok?

File: types/result.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func (r TransactionResult) Reverted() bool {
 // TransactionResultDebug provides details about unsuccessful transaction execution
 type TransactionResultDebug struct {
 	Message string
-	Meta    map[string]string
+	Meta    map[string]any
```
Comment (janezpodhostnik) [inline types/result.go:60]: does this have to be any?

**Review by janezpodhostnik — COMMENTED** [review]
Looks good. Nice change. Had some questions.

File: server/storage.go (hunk 1 above)
Comment (bluesign) [reply server/storage.go:130]: @janezpodhostnik badger is retiring soon ( #301 ), that's why I skipped this one.

File: types/result.go (hunk 2 above)
Comment (bluesign) [reply types/result.go:60]: @janezpodhostnik to use it directly with logger, I think it has to be any ( otherwise logger is skipping somehow )

**Comment by turbolent** [comment]
What are the advantages?

**Comment by bluesign** [comment]
> What are the advantages?

- logrus is in maintenance mode 

> I believe Logrus' biggest contribution is to have played a part in today's widespread use of structured logging in Golang. There doesn't seem to be a reason to do a major, breaking iteration into Logrus V2, since the fantastic Go community has built those independently. Many fantastic alternatives have sprung up. Logrus would look like those, had it been re-designed with what we know about structured logging in Go today. Check out, for example, [Zerolog](https://github.com/rs/zerolog), [Zap](https://github.com/uber-go/zap), and [Apex](https://github.com/apex/log).

- fvm already using zerolog, with this we can use the same logger for tracing fvm. ( gas usage response, real-time cadence logging (now we show logs after execution ) )

tbh my main use case was fvm compatibility in logging.

**Comment by turbolent** [comment]
Oh, I had no idea, that alone is a great reason to migrate

**Review by janezpodhostnik — APPROVED** [review]

**Comment by sideninja** [comment]
@bluesign can you take a look at the conflicts and we can merge later.

**Pull request merged**